src/main/scala/codacy/plugins/test/SourceTests.scala

```diff
@@ -0,0 +1,38 @@
+package codacy.plugins.test
+
+import java.nio.file.Path
+
+import codacy.plugins.docker.{DockerPlugin, Pattern}
+import codacy.utils.{FileHelper, Printer}
+
+import scala.util.Properties
+
+object SourceTests extends ITest{
+  override val opt: String = "source"
+
+  override def run(plugin: DockerPlugin, testSources: Seq[Path], dockerImageName: String, optArgs: Seq[String]): Boolean = {
+    Printer.green(s"Running SourceTests:")
+    DockerHelpers.withDocsDirectory(dockerImageName) { baseDocDir =>
+    plugin.spec match{
+      case Some(spec) =>
+        val patterns = DockerHelpers.toPatterns(spec)
+        patterns.forall{ case pattern: Pattern =>
```

Comment:
Any reason for us to have the : Pattern or we could just have the case pattern => ?